src/dmqnode/storage/engine/overflow/file/IndexFile.d

```diff
@@ -71,11 +71,28 @@ class IndexFile: PosixFile
 
     public FILE* stream;
 
+    /***************************************************************************
+
+        A flag for the invariant so that `stream` is checked only after the
+        constructor has returned. Otherwise the check fails when the super
+        constructor calls a protected method.
+
+    ***************************************************************************/
+
+    version (D_Version2)
+        mixin("version(assert) private bool ctor_returned = false;");
```

Comment:
`open` can't be overriden as private?

Reply:
`HeadTruncationTestFile` overrides it. Is it possible to override `private` methods in D2?

Reply:
No, I thought this class is the last one in the derive chain (it is possible to override `protected`/`public` method as `private` but it is final after that). Just trying to come up with less hacky solution for this case :)

Reply:
```d
class A { void f(){} }
class B { private override void f(){} }
```
` Error: function overr.B.f private method is not virtual and cannot override`

Reply:
I stand corrected. Weird that it doesn't simply interpret it as `final` here.